pkg/controller/flow/reconcile.go

```diff
@@ -0,0 +1,360 @@
+		return nil, err
+	}
+
+	// Should make sure channel is what it should be. For now, just assume it's fine
+	// if it exists.
+	return channel, err
+}
+
+func (r *reconciler) createChannel(flow *v1alpha1.Flow) (*channelsv1alpha1.Channel, error) {
+	channelName := flow.Name
+	channel := &channelsv1alpha1.Channel{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      channelName,
+			Namespace: flow.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*r.NewControllerRef(flow),
+			},
+		},
+		Spec: channelsv1alpha1.ChannelSpec{
+			ClusterBus: defaultBusName,
+		},
+	}
+	if err := r.client.Create(context.TODO(), channel); err != nil {
+		return nil, err
+	}
+	return channel, nil
+}
+
+func (r *reconciler) reconcileSubscription(channelName string, target string, flow *v1alpha1.Flow) (*channelsv1alpha1.Subscription, error) {
+	subscriptionName := flow.Name
+
+	subscription := &channelsv1alpha1.Subscription{}
+	err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: flow.Namespace, Name: subscriptionName}, subscription)
+	if errors.IsNotFound(err) {
+		subscription, err = r.createSubscription(channelName, target, flow)
+		if err != nil {
+			glog.Errorf("Failed to create subscription %q : %v", subscriptionName, err)
+			return nil, err
+		}
+	} else if err != nil {
+		glog.Errorf("Failed to reconcile subscription %q failed to get subscriptions : %v", subscriptionName, err)
+		return nil, err
+	}
+
+	// Should make sure subscription is what it should be. For now, just assume it's fine
+	// if it exists.
+	return subscription, err
+}
+
+func (r *reconciler) createSubscription(channelName string, target string, flow *v1alpha1.Flow) (*channelsv1alpha1.Subscription, error) {
+	subscriptionName := flow.Name
+	subscription := &channelsv1alpha1.Subscription{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      subscriptionName,
+			Namespace: flow.Namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				*r.NewControllerRef(flow),
+			},
+		},
+		Spec: channelsv1alpha1.SubscriptionSpec{
+			Channel:    channelName,
+			Subscriber: target,
+		},
+	}
+	if err := r.client.Create(context.TODO(), subscription); err != nil {
+		return nil, err
+	}
+	return subscription, nil
+}
+
+func (r *reconciler) reconcileFeed(channelDNS string, flow *v1alpha1.Flow) (*feedsv1alpha1.Feed, error) {
+	feedName := flow.Name
+
+	feed := &feedsv1alpha1.Feed{}
+	err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: flow.Namespace, Name: feedName}, feed)
+	if errors.IsNotFound(err) {
+		feed, err = r.createFeed(channelDNS, flow)
+		if err != nil {
+			glog.Errorf("Failed to create feed %q : %v", feedName, err)
+			return nil, err
+		}
+	} else if err != nil {
+		glog.Errorf("Failed to reconcile feed %q failed to get feeds : %v", feedName, err)
+		return nil, err
+	}
+
+	glog.Infof("Reconciled feed: %+v", feed)
+	// Should make sure feed is what it should be. For now, just assume it's fine
+	// if it exists.
+	return feed, err
```

Comment:
Seems like `err` can only be `nil` here, so it might be clearer to `return channel, nil`.

Reply:
done